modules/pxp-module-puppet

```diff
@@ -282,6 +282,10 @@ def action_run(input)
         1
       end
 
+      # flush the stdout/stderr before writing the exitcode
+      # file to avoid pxp-agent reading incomplete output
+      $stdout.fsync
```

Comment:
I don't think it makes much sense, as it doesn't provide any string guarantees, it only guarantees ruby internal buffers are flushed on each write, but it doesn't guarantee the same for OS level buffers - so we'd still need the ```fsync``` call (which does call the system fsync, thus ensuring any buffered data is written to disk).